### PR TITLE
Reward page

### PR DESCRIPTION
**Pull request by cindy1u0**
Now you can pop balloons. (This is yet connected to the end of assessment. Not sure if I should include that in this PR)

Screen Shot 2021-01-31 at 3 06 57 AM
Screen Shot 2021-01-31 at 3 06 44 AM
Screen Shot 2021-01-31 at 3 06 35 AM

### CODE REVIEW

File: frontend/src/components/Balloon.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  ReactElement,
+  useState,
+} from "react";
+import styled from "styled-components";
+import { BALLOON } from "../constants/images";
+
+interface BallonProps {
+  top?: number;
+  left?: number;
+  onPop?: any;
```
Comment (cindy1u0) [inline frontend/src/components/Balloon.tsx]: before I merge if I get approved, I honestly don't know what to put for this. I tried doing either void or () => void, both gave an error in either Balloon.tsx or Reward.tsx...

Comment (jrosen081) [reply frontend/src/components/Balloon.tsx]: I think you should have it as `(void) => void;`. I _thought_ `() => void;` would work, but I certainly could be wrong

File: frontend/src/components/Balloon.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+  @media (max-width: 600px) {
+    transform: scale(0.4);
+  }
+`;
+
+const Container = styled.div`
+  position: absolute;
+  top: ${(props: BallonProps) => props.top}%;
+  left: ${(props: BallonProps) => props.left}%;
```
Comment (jrosen081) [inline frontend/src/components/Balloon.tsx]: Can we destructure these please?

File: frontend/src/pages/Interventions/Reward.tsx [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    <Layout>
+      <Container>
+        <TitleContainer>
+          pop all the balloons to go back to Dashboard
+        </TitleContainer>
+        <CloudGroup />
+        {score === 5 ? (
+          <TitleContainer>
+            Congradulations! You may go back to Dashboard now!
+          </TitleContainer>
```
Comment (jrosen081) [inline frontend/src/pages/Interventions/Reward.tsx]: I think we should probably automatically redirect them. So like have this on the screen for a second and then redirect?

**Review by jrosen081 — APPROVED** [review]
This is looking really good. I think this is supposed to be after interventions, so if you want to hook it up in this pr, that could be good!

File: frontend/src/pages/Interventions/Reward.tsx [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+        ) : (
+          <ComponentContainer>
+            <Balloon
+              top={20}
```
Comment (stefan-philip) [inline frontend/src/pages/Interventions/Reward.tsx]: Should we randomize the positions of the balloons?

**Review by stefan-philip — APPROVED** [review]

File: frontend/src/pages/Interventions/Reward.tsx (hunk 4 above)
Comment (jrosen081) [reply frontend/src/pages/Interventions/Reward.tsx]: @stephmiano What do you think? Should we randomize the positions?

Comment (stephmiano) [reply frontend/src/pages/Interventions/Reward.tsx]: yes i think randomizing them would be good

**Comment by cindy1u0** [comment]
The revision had randomization (using a library for seeding) and also redirecting back to dashboard after 2 seconds (can be changed)

File: frontend/src/pages/Reward/RewardPage.tsx [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+  return (
+    <Layout>
+      <Container>
+        <TitleContainer>
+          pop all the balloons to go back to Dashboard
+        </TitleContainer>
+        <CloudGroup />
+        {score === maxBalloon ? (
+          <TitleContainer>
+            Congradulations! You have popped all the balloons :)
```
Comment (stefan-philip) [inline frontend/src/pages/Reward/RewardPage.tsx]:
```suggestion
            Congratulations! You have popped all the balloons :)
```

typo

File: frontend/src/App.tsx [hunk 6]
```diff
@@ -23,10 +24,26 @@ const App = (): ReactElement => {
...
+          exact
+          path="/interventions/:id"
+          component={Activities}
+        />
+        <AuthenticatedRoute
+          exact
+          path="/assessments/:id"
+          component={Quiz}
+        />
+        <AuthenticatedRoute path="/reward" component={Reward} />
```
Comment (jrosen081) [inline frontend/src/App.tsx]: I think this path should be "/interventions/reward", to be clear what the reward is for.

File: frontend/src/pages/Reward/RewardPage.tsx [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+  const randomNum = (seedNum: number, min: number, max: number) => {
+    let seed = seedrandom(`${seedNum * randSeed}`);
+    return seed() * (max - min) + min;
+  };
+
+  const BallonGroups = () => {
+    let balloons = [];
+    for (let i = 0; i < 5; i++) {
+      let t = randomNum(i, 20, 80);
+      let l = randomNum(i + 1, 10, 60);
```
Comment (jrosen081) [inline frontend/src/pages/Reward/RewardPage.tsx:59]: I wonder if we should use the actual dimensions of the screen?

**Review by jrosen081 — APPROVED** [review]
This looks great to me! If you could also hook this up to interventions that would be perfect!

**Review by stefan-philip — APPROVED** [review]
Great work!!

File: frontend/src/pages/Reward/RewardPage.tsx (hunk 5 above)
Comment (cindy1u0) [reply frontend/src/pages/Reward/RewardPage.tsx]: yikes can't type

File: frontend/src/pages/Reward/RewardPage.tsx (hunk 7 above)
Comment (cindy1u0) [reply frontend/src/pages/Reward/RewardPage.tsx:59]: ah I tried with the actual dimensions, and some of the smaller ones look super weird lol

File: frontend/src/App.tsx (hunk 6 above)
Comment (cindy1u0) [reply frontend/src/App.tsx]: Ah should I do this part with this week ticket -> hooks this up to assessment

Comment (jrosen081) [reply frontend/src/App.tsx]: Idk it is a super quick change that could be done in this pr tbh.

Comment (cindy1u0) [reply frontend/src/App.tsx]: yessir ill change it xD

**Comment by cindy1u0** [comment]
aight final push

**Pull request merged**